Feat: Umbrella Activation (#715)
bgd-labs/aave-proposals-v3: src/20250515_AaveV3Ethereum_UmbrellaActivation/UmbrellaActivation.md

1. **Umbrella Activation**
2. **Legacy stk tokens adjustment**
3. **Slashing robot activation**

### 1. **Umbrella Activation**

This payload handles the deployment and configuration of the new Umbrella system for deficit elimination and staking.

#### Contract registration

- Registers the `Umbrella` contract in the `POOL_ADDRESSES_PROVIDER`.
  - Enables the `Umbrella` to be used for reserve deificit elimination.

#### Reserve Deficit Elimination

- Eliminates existing reserve deficits for the following assets: USDC, USDT, WETH and GHO.
  - Calculates current reserve deficits amounts.
  - Transfers the required amounts from the `Collector` to the `Executor` contract.
  - Executes deficit elimination using the transferred funds.

#### Stk Tokens Creation

- Creates `stk` tokens for the above assets:
  - `stkwaUSDC.V1`, `stkwasUSDT.V1`, stkwaWETH.V1` are backed by their respective stata tokens.
  - `stkGHO.V1` is backed by the GHO underlying token.

#### Reward Configuration

- Sets up reward emissions for each `stk` token:
  - `stkwaUSDC.V1` -> `aUSDC`, `stkwaUSDT.V1` -> `aUSDT`, `stkwaWETH.V1` -> `aWETH`
  - `stkGHO.V1` -> `GHO`
  - For each:

>     - `maxEmissionPerSecond` and `targetLiquidity` are configured using values from the forum.

Reviewer: ```suggestion
    - `maxEmissionPerSecond` and `targetLiquidity` are configured using values from the governance discussion, check the [summary table](#summary-table) for more details
```

    - `distributionEnd` is set to 1 year from the execution.
    - `rewardPayer` is st to the `Collector`.

#### Slashing Configuration

- Sets up `SlashingConfig` for each `stk` token:
  - `reserve`: Corresponds to the base token (e.g., `USDT` for `stkwaUSDT.V1`, `GHO` for `stkGHO.V1`).
  - `liquidationFee`: Set to zero for all stk tokens.

>   - `deficitOffset`: Values sourced from the forum.

Reviewer: ```suggestion
  - `deficitOffset`: Values sourced from the governance discussion.
```

#### Stk Tokens Creation

- Creates `stk` tokens for the above assets:
  - `stkwaUSDC.V1`, `stkwaUSDT.V1`, `stkwaWETH.V1` are backed by their respective stata tokens.
  - `stkGHO.V1` is backed by the GHO underlying token.

#### Reward Configuration

- Sets up reward emissions for each `stk` token:
  - `stkwaUSDC.V1` -> `aUSDC`, `stkwaUSDT.V1` -> `aUSDT`, `stkwaWETH.V1` -> `aWETH`
  - `stkGHO.V1` -> `GHO`
  - For each:
    - `maxEmissionPerSecond` and `targetLiquidity` are configured using values from the governance discussion, check the [summary table](#summary-table) for more details
    - `distributionEnd` is set to 1 year from the execution.
    - `rewardPayer` is st to the `Collector`.

#### Slashing Configuration

- Sets up `SlashingConfig` for each `stk` token:
  - `reserve`: Corresponds to the base token (e.g., `USDT` for `stkwaUSDT.V1`, `GHO` for `stkGHO.V1`).
  - `liquidationFee`: Set to zero for all stk tokens.
  - `deficitOffset`: Values sourced from the governance discussion.
  - `umbrellaStakeUnderlyingOracle`:
    - For stata-based tokens: the stata token itself.
    - For `GHO`: a `GHO` oracle with a mocked price.

> #### Role and Permisssion Management

Reviewer 2: ```suggestion
#### Role and Permission Management
```

|    waUSDC    |     USDC      |     66_000_000     |  2_330_000   |     20 days / 2 days      |    100_000     |
|    waUSDT    |     USDT      |    104_000_000     |  3_670_000   |     20 days / 2 days      |    100_000     |
|    waWETH    |     WETH      |       25_000       |     550      |     20 days / 2 days      |       50       |
|     GHO      |      GHO      |     12_000_000     |  1_200_000   |     20 days / 2 days      |    100_000     |

- Target Liquidity is denominated in the contracts in wrapped aTokens, increasing over time in exchange rate. That means the Target Liquidity itself will grow slightly over time and rewards will need to be periodically adjusted. For the sake of simplicity, the number on the table is in equivalent terms of underlying (USDC, USDT, WETH), not in wrapped aTokens.

### 2. **Legacy stk tokens adjustment**

This payload introduces updates to the legacy Safety Module staking tokens to align with the activation of the new Umbrella system.

#### Emission Reductions

- Reduces (`AAVE`/day) reward emissions for the following tokens:
  - `stkGHO`: 100 -> 0

>   - `stkAAVE`: 360 -> 316

Reviewer 3: ```suggestion
  - `stkAAVE`: 360 -> 315
```
